connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTest.java

```diff
@@ -804,6 +833,63 @@ public void testConverterOverrides() throws Exception {
         PowerMock.verifyAll();
     }
 
+    @Test
+    public void testProducerConfigsWithoutOverrides() {
+        expectConverters();
+        expectStartStorage();
+        PowerMock.replayAll();
+        worker = new Worker(WORKER_ID, new MockTime(), plugins, config, offsetBackingStore);
+        assertEquals(defaultProducerConfigs, worker.producerConfigs(EasyMock.mock(ConnectorConfig.class), new ConnectorTaskId("test", 1)));
+    }
+
+    @Test
+    public void testProducerConfigsWithOverrides() {
```

Comment:
I'm fine with these as is, but you could also change the methods to be `static` in `Worker` and accept the `WorkerConfig` as a parameter since that's the only class member they use. Would get rid of all the distracting mocks and expect calls and focus the tests on the key functionality of those methods.